dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java

```diff
@@ -420,25 +418,27 @@ void shouldMapOrgUnitWhenProgramProvidedAndRequestedOrgUnitInSearchScope(
       OrganisationUnitSelectionMode orgUnitMode, AccessLevel accessLevel)
       throws ForbiddenException, BadRequestException {
     Program program = new Program();
+    program.setUid(CodeGenerator.generateUid());
     program.setAccessLevel(accessLevel);
 
-    OrganisationUnit searchScopeOrgUnit = createOrgUnit("searchScopeOrgUnit", "uid4");
-    OrganisationUnit searchScopeChildOrgUnit = createOrgUnit("searchScopeChildOrgUnit", "uid5");
+    OrganisationUnit searchScopeOrgUnit = createOrgUnit("searchScopeOrgUnit");
+    OrganisationUnit searchScopeChildOrgUnit = createOrgUnit("searchScopeChildOrgUnit");
     searchScopeOrgUnit.setChildren(Set.of(searchScopeChildOrgUnit));
     searchScopeChildOrgUnit.setParent(searchScopeOrgUnit);
 
     User user = new User();
+    user.setUid(CodeGenerator.generateUid());
     user.setUsername("testB");
-    user.setOrganisationUnits(Set.of(createOrgUnit("captureScopeOrgUnit", "uid")));
+    user.setOrganisationUnits(Set.of(createOrgUnit("captureScopeOrgUnit")));
     user.setTeiSearchOrganisationUnits(Set.of(searchScopeOrgUnit));
 
     when(organisationUnitService.getOrganisationUnit(searchScopeChildOrgUnit.getUid()))
         .thenReturn(searchScopeChildOrgUnit);
 
     EventOperationParams operationParams =
         eventBuilder
-            .programUid(program.getUid())
-            .orgUnitUid(searchScopeChildOrgUnit.getUid())
+            .program(UID.of(program))
+            .orgUnit(UID.of(searchScopeChildOrgUnit))
```

Comment:
Would it make the API easier to use if we allowed passing things implementing the UidObject interface and then create the UIO ourselves? or do we then risk of not getting validated UIDs as the UID type throws if its an invalid UID while the interface might not be able guarantee that given that its

```java
public interface UidObject {
  /**
   * @return external unique ID of the object as used in the RESTful API
   */
  @JsonProperty(value = "id")
  String getUid();
}
```

Reply:
It would make it easier for tests for sure but in the implementation, at least from the exporter endpoints, at this point we do not have objects but just the UIDs passed as parameters.
For some specific cases like, currentUser, that could be useful.
The interface does not guarantee that the UID is valid, but we are kind of living with that assumption anyway.
I am gonna check how it would look like in the way you described.
For example, for the SMS part, would be easier to create the params.